src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.cpp

```diff
@@ -67,308 +64,207 @@ namespace argos {
-         Left = cLeftXY;
-         Left *= Cos(c_angle);
-         Left -= cForwardXY * Sin(c_angle);
-         Left.Normalize();
-         
-         /* To calculate the new Up vector, a cross-product is enough */
-         Up = Forward;
-         Up.CrossProduct(Left).Normalize();
-      }
-   }
-
-   /****************************************/
-   /****************************************/
-
-   void CQTOpenGLCamera::SPlacement::Translate(const CVector3& c_delta) {
-      Position += Forward * c_delta.GetX() + Left * c_delta.GetY() + Up * c_delta.GetZ();
-      Target = Position;
-      Target += Forward;
-   }
-
-   /****************************************/
-   /****************************************/
-
-   void CQTOpenGLCamera::SPlacement::Do() {
-      gluLookAt(
-         Position.GetX(),
-         Position.GetY(),
-         Position.GetZ(),
-         Target.GetX(),
-         Target.GetY(),
-         Target.GetZ(),
-         Up.GetX(),
-         Up.GetY(),
-         Up.GetZ());
-   }
-
-   /****************************************/
-   /****************************************/
-
    void CQTOpenGLCamera::SPlacement::CalculateYFieldOfView() {
       YFieldOfView = ToDegrees(2.0f * ATan2(0.027f * 0.5f, LensFocalLength));
    }
 
    /****************************************/
    /****************************************/
 
-   void CQTOpenGLCamera::SPlacement::CalculateSensitivity() {
-      MotionSensitivity = MOVE_GAIN * ::exp(LensFocalLength);
-      RotationSensitivity = ROTATE_GAIN * ::exp(-LensFocalLength);
-   }
-
-   /****************************************/
-   /****************************************/
-
-   void CQTOpenGLCamera::STimelineItem::Init(TConfigurationNode& t_tree) {
-      GetNodeAttribute(t_tree, "idx", Idx);
-      if(Idx > 11) {
-         THROW_ARGOSEXCEPTION("Error initializing QTOpenGL camera timeline: value given for 'idx' is out of bounds. Value = \"" << Idx << "\", allowed [0-11].");
-      }
-      GetNodeAttribute(t_tree, "step", Step);
-   }
-
-   /****************************************/
-   /****************************************/
-
    CQTOpenGLCamera::CQTOpenGLCamera() :
-      m_unActivePlacement(0) {
+      m_bEnableTimeline(false),
+      m_unLoop(0) {
    }
 
    /****************************************/
    /****************************************/
 
-   CQTOpenGLCamera::~CQTOpenGLCamera() {
-      while(!m_listTimeline.empty()) {
-         delete m_listTimeline.back();
-         m_listTimeline.pop_back();
-      }
-   }
+   CQTOpenGLCamera::~CQTOpenGLCamera() {}
    
    /****************************************/
    /****************************************/
 
    void CQTOpenGLCamera::Init(TConfigurationNode& t_tree) {
-      if(NodeExists(t_tree, "camera")) {
-         try {
-            TConfigurationNode tCameraNode;
-            tCameraNode = GetNode(t_tree, "camera");
-            /* Parse "placement" nodes */
+      try {
+         if(NodeExists(t_tree, "placements")) {
```

Comment:
This function seems too long. Consider breaking into smaller functions for improved readability:

- Parsing the placements
- Parsing the timeline nodes

Reply:
I wouldn't break this into separate functions. I think it is a bit of an anti-pattern to break a function down into smaller functions when they are only going to be called from a exactly one call site.

Reply:
I agree, function length alone is not a good metric to make the structure more complex.